osu.Game/Beatmaps/BeatmapInfo.cs

```diff
@@ -152,6 +152,25 @@ public string StoredBookmarks
         [JsonIgnore]
         public DifficultyRating DifficultyRating => BeatmapDifficultyCache.GetDifficultyRating(StarDifficulty);
 
+        /// <summary>
+        /// Creates a new <see cref="BeatmapInfo"/> with initialised defaults.
+        /// </summary>
+        /// <remarks>
+        /// The defaults applied below are not applied directly to <see cref="BeatmapInfo"/> members,
```

Comment:
Does this imply that the `BeatmapInfo` defaults are made to match legacy settings? Seems a bit back to front if so - I would much rather those defaults were applied in the `LegacyBeatmapDecoder` rather than hard coded into a non-legacy class.

Reply:
Pretty sure that's how things stand right now, yeah. Am fine with moving to decoder, but probably in a separate pull? (can come before/as prereq to this one if deemed required)

In general the legacy encoder will sometimes skip writing some configuration keys rather than write out the default, which is why this avenue was taken, I believe.